packages/core/src/createActor.ts

```diff
@@ -491,7 +491,26 @@ export class Actor<TLogic extends AnyActorLogic>
         // TODO: rethink cleanup of observers, mailbox, etc
         return this;
       case 'error':
+        // in this case, creation of the initial snapshot caused an error.
+
+        // **if the actor has no observer when start() is called**, this error would otherwise be
+        // thrown to the global error handler. to prevent this--and allow start() to be wrapped in try/catch--we
+        // create a temporary observer that receives the error (via _reportError()) and rethrows it from this call stack.
```

Comment:
I'm wondering if we can do this a different way without creating a temp sub

Reply:
@davidkpiano The folded section [in this comment](https://github.com/statelyai/xstate/issues/4928#issuecomment-2294458587) has an alternate proposal, which I explored but didn't get working fully.